lib/state/analytics/middleware.ts

```diff
@@ -0,0 +1,82 @@
+import analytics from '../../analytics';
+import actions from '../actions';
+
+import * as A from '../action-types';
+import * as S from '../';
+import * as T from '../../types';
+
+export const middleware: S.Middleware = store => {
+  return next => (action: A.ActionType) => {
+    const result = next(action);
```

Comment:
it should be handled in our analytics code itself, but this is a great place to check for the analytics toggle and skip the middleware if analytics are turned off

```ts
const result = next( action );

if ( ! window.sendAnalytics ) {
	return result;
}
```